src/firebolt/async_db/cursor.py

```diff
@@ -372,7 +389,7 @@ async def _do_execute(
                     Optional[List[List[RawColType]]],
                 ] = (-1, None, None, None)
                 if isinstance(query, SetParameter):
-                    # Validate parameter by executing simple query with it
+                    # Validate parameter by executing simple query with it.
                     resp = await self._api_request(
```

Comment:
Check if user selected `async_execution` here, and warn that since it's a set statement, it will still be executed synchronously

Reply:
Since you said "warn," I did this via a log statement, but maybe we should actually error out. Later calls to `cancel()` or `get_query_status()` will fail.

Reply:
Actually, and the return type from all the execute fns will be wrong.

Reply:
This now errors out and a test is included in `test_cursor.py`.